Source/BugsnagNotifier.m

```diff
@@ -81,40 +82,41 @@
  *
  *  @param writer report writer which will receive updated metadata
  */
-void BSSerializeDataCrashHandler(const BSG_KSCrashReportWriter *writer) {
-    if (bsg_g_bugsnag_data.configJSON) {
-        writer->addJSONElement(writer, "config", bsg_g_bugsnag_data.configJSON);
-    }
-    if (bsg_g_bugsnag_data.metaDataJSON) {
-        writer->addJSONElement(writer, "metaData",
-                               bsg_g_bugsnag_data.metaDataJSON);
-    }
+void BSSerializeDataCrashHandler(const BSG_KSCrashReportWriter *writer, int type) {
+    BOOL userReported = BSG_KSCrashTypeUserReported == type;
 
     if (hasRecordedSessions) { // a session is available
         // persist session info
         writer->addStringElement(writer, "id", (const char *) sessionId);
         writer->addStringElement(writer, "startedAt", (const char *) sessionStartDate);
         writer->addUIntegerElement(writer, "handledCount", handledCount);
 
-        if (!bsg_g_bugsnag_data.handledState) {
+        if (!userReported) {
             writer->addUIntegerElement(writer, "unhandledCount", 1);
         } else {
             writer->addUIntegerElement(writer, "unhandledCount", 0);
         }
     }
-
-    if (bsg_g_bugsnag_data.handledState) {
-        writer->addJSONElement(writer, "handledState",
-                               bsg_g_bugsnag_data.handledState);
+    if (bsg_g_bugsnag_data.configJSON) {
+        writer->addJSONElement(writer, "config", bsg_g_bugsnag_data.configJSON);
     }
-
     if (bsg_g_bugsnag_data.stateJSON) {
         writer->addJSONElement(writer, "state", bsg_g_bugsnag_data.stateJSON);
     }
-    if (bsg_g_bugsnag_data.userOverridesJSON) {
-        writer->addJSONElement(writer, "overrides",
-                               bsg_g_bugsnag_data.userOverridesJSON);
+
+    // write additional user-supplied metadata
+    if (userReported) {
+        if (bsg_g_bugsnag_data.metaDataJSON) {
+            writer->addJSONElement(writer, "metaData", bsg_g_bugsnag_data.metaDataJSON);
```

Comment:
This value should be attached to all reports as it contains the values from [`+[Bugsnag addAttribute:withValue:toTabWithName:]`](https://docs.bugsnag.com/platforms/ios/customizing-error-reports/#adding-custom-diagnostics). We should add a test for this case.